code/modules/organs/subtypes/machine.dm

```diff
@@ -62,26 +62,99 @@
 	desc = "A small, powerful cell for use in fully prosthetic bodies."
 	icon = 'icons/obj/power.dmi'
 	icon_state = "scell"
-	organ_tag = "cell"
+	organ_tag = BP_CELL
 	parent_organ = BP_CHEST
-	vital = 1
-	var/emp_counter = 0
+	vital = TRUE
+	max_damage = 80
+	relative_size = 80
+	var/open = FALSE
+	var/obj/item/cell/cell = /obj/item/cell/super
+	//at 0.8 completely depleted after 60ish minutes of constant walking or 130 minutes of standing still
+	var/servo_cost = 0.8
 
 /obj/item/organ/internal/cell/Initialize()
 	robotize()
+	cell = new cell(src)
 	. = ..()
 
+/obj/item/organ/internal/cell/proc/percent()
+	if(!cell)
+		return 0
+	return get_charge()/cell.maxcharge * 100
+
+/obj/item/organ/internal/cell/proc/get_charge()
+	if(!cell)
+		return 0
+	if(status & ORGAN_DEAD)
+		return 0
+	return round(cell.charge*(1 - damage/max_damage))
+
+/obj/item/organ/internal/cell/proc/checked_use(var/amount)
+	if(!is_usable() || !cell)
+		return FALSE
+	return cell.checked_use(amount)
+
+/obj/item/organ/internal/cell/proc/use(var/amount)
+	if(!is_usable() || !cell)
+		return
+	return cell.use(amount)
+
 /obj/item/organ/internal/cell/process()
 	..()
-	if(emp_counter)
-		emp_counter--
+	if(!owner)
+		return
+	if(owner.stat == DEAD)	//not a drain anymore
+		return
+	var/cost = get_power_drain()
+	if(world.time - owner.l_move_time < 15)
+		cost *= 2
+	if(!checked_use(cost) && owner.isSynthetic())
+		if(!owner.lying && !owner.buckled)
+			to_chat(owner, "<span class='warning'>You don't have enough energy to function!</span>")
+		owner.Paralyse(3)
+
+/obj/item/organ/internal/cell/proc/get_power_drain()	
+	return servo_cost
 
 /obj/item/organ/internal/cell/emp_act(severity)
-	emp_counter += 30/severity
-	if(emp_counter >= 30)
-		owner.Paralyse(emp_counter/6)
-		to_chat(owner, "<span class='danger'>%#/ERR: Power leak detected!$%^/</span>")
+	..()
+	if(cell)
+		cell.emp_act(severity)
+
+/obj/item/organ/internal/cell/attackby(obj/item/W, mob/user)
+	if(isscrewdriver(W))
+		if(open)
+			open = FALSE
+			to_chat(user, SPAN_NOTICE("You screw the battery panel in place."))
+		else
+			open = TRUE
+			to_chat(user, SPAN_NOTICE("You unscrew the battery panel."))
+
+	if(iscrowbar(W))
+		if(open)
+			if(cell)
+				user.put_in_hands(cell)
+				to_chat(user, SPAN_NOTICE("You remove \the [cell] from \the [src]."))
+				cell = null
+
+	if (istype(W, /obj/item/cell))
+		if(open)
+			if(cell)
+				to_chat(user, SPAN_WARNING("There is a power cell already installed."))
+			else if(user.unEquip(W, src))
+				cell = W
+				to_chat(user, SPAN_NOTICE("You insert \the [cell].</span>"))
+
+/obj/item/organ/internal/cell/replaced()
+	..()
+	// This is very ghetto way of rebooting an IPC. TODO better way.
+	if(owner && owner.stat == DEAD)
+		owner.stat = CONSCIOUS
```

Comment:
This is just bringing back code that was removed a long time ago. It is really buggy and would result in people helping to get revive after.

Reply:
removed